fedimint-core/src/task/jit.rs

```diff
@@ -103,30 +115,60 @@ where
         Self {
             _cancel_tx: cancel_tx,
             val_rx: Arc::new(Some(val_rx).into()),
-            val: sync::OnceCell::new(),
+            val: sync::OnceCell::new().into(),
         }
     }
 
     /// Get the reference to the value, potentially blocking for the
     /// initialization future to complete
     pub async fn get_try(&self) -> std::result::Result<&T, OneTimeError<E>> {
-        #[allow(clippy::await_holding_lock)]
+        /// Temporarily taken data out of `arc`, that will be put back on drop,
+        /// unless canceled
+        ///
+        /// Used to achieve cancelation safety.
```

Comment:
Not sure if I understand.

Before I was holding the `std::Mutex` over `await` which was OK because contention is not possible there, but I discovered it is making the future not `Send`, so I had to remove it, but that requires taking care of cancelation safety.